From b3c07c884371b7b535a6daf236350800937836e8 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Sun, 15 Jul 2018 20:10:07 +0200
Subject: [PATCH 08/12] LineageParts: Remove center clock position for devices
 with notch

* Also fix quick pulldown entries when in RTL

Change-Id: I2861a478d11b9671e1a98cb8e5c223481e728a83
---
 res/values/arrays.xml                         | 32 +++++++++++++++++++
 .../statusbar/StatusBarSettings.java          | 15 ++++++++-
 2 files changed, 46 insertions(+), 1 deletion(-)

diff --git a/res/values/arrays.xml b/res/values/arrays.xml
index a0182a6..79b0149 100644
--- a/res/values/arrays.xml
+++ b/res/values/arrays.xml
@@ -228,6 +228,32 @@
         <item>2</item>
     </string-array>
 
+    <string-array name="status_bar_clock_position_values_rtl" translatable="false">
+        <item>2</item>
+        <item>1</item>
+        <item>0</item>
+    </string-array>
+
+    <string-array name="status_bar_clock_position_entries_notch" translatable="false">
+        <item>@string/status_bar_clock_position_right</item>
+        <item>@string/status_bar_clock_position_left</item>
+    </string-array>
+
+    <string-array name="status_bar_clock_position_entries_notch_rtl" translatable="false">
+        <item>@string/status_bar_clock_position_left</item>
+        <item>@string/status_bar_clock_position_right</item>
+    </string-array>
+
+    <string-array name="status_bar_clock_position_values_notch" translatable="false">
+        <item>0</item>
+        <item>2</item>
+    </string-array>
+
+    <string-array name="status_bar_clock_position_values_notch_rtl" translatable="false">
+        <item>2</item>
+        <item>0</item>
+    </string-array>
+
     <string-array name="status_bar_am_pm_entries" translatable="false">
         <item>@string/status_bar_am_pm_normal</item>
         <item>@string/status_bar_am_pm_small</item>
@@ -287,6 +313,12 @@
         <item>2</item>
     </string-array>
 
+    <string-array name="status_bar_quick_qs_pulldown_values_rtl" translatable="false">
+        <item>2</item>
+        <item>1</item>
+        <item>0</item>
+    </string-array>
+
     <!-- Weather temperature -->
     <string-array name="weather_temp_unit_entries" translatable="false">
         <item>@string/weather_settings_temp_celsius</item>
diff --git a/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java b/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
index 055fddc..bf4bcc6 100644
--- a/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
+++ b/src/org/lineageos/lineageparts/statusbar/StatusBarSettings.java
@@ -80,10 +80,23 @@ public class StatusBarSettings extends SettingsPreferenceFragment
     public void onResume() {
         super.onResume();
 
+        boolean hasNotch = getResources().getBoolean(
+                org.lineageos.platform.internal.R.bool.config_haveNotch);
+
         // Adjust status bar preferences for RTL
         if (getResources().getConfiguration().getLayoutDirection() == View.LAYOUT_DIRECTION_RTL) {
-            mStatusBarClock.setEntries(R.array.status_bar_clock_position_entries_rtl);
+            if (hasNotch) {
+                mStatusBarClock.setEntries(R.array.status_bar_clock_position_entries_notch_rtl);
+                mStatusBarClock.setEntryValues(R.array.status_bar_clock_position_values_notch_rtl);
+            } else {
+                mStatusBarClock.setEntries(R.array.status_bar_clock_position_entries_rtl);
+                mStatusBarClock.setEntryValues(R.array.status_bar_clock_position_values_rtl);
+            }
             mQuickPulldown.setEntries(R.array.status_bar_quick_qs_pulldown_entries_rtl);
+            mQuickPulldown.setEntryValues(R.array.status_bar_quick_qs_pulldown_values_rtl);
+        } else if (hasNotch) {
+            mStatusBarClock.setEntries(R.array.status_bar_clock_position_entries_notch);
+            mStatusBarClock.setEntryValues(R.array.status_bar_clock_position_values_notch);
         }
     }
 
-- 
2.17.1

